src/components/Header/useScreenWidth.ts

```diff
@@ -0,0 +1,25 @@
+import { useState, useEffect } from "react";
+
+const useScreenWidth = () => {
+  const [isMobile, setIsMobile] = useState(false);
```

Comment:
tak myślę, że warto zrobić z tego bardziej generyczny hook i zamiast 'isMobile' i 'setIsMobile' użyć nazw, które nie będą dotyczyć specyficznego breakpointa - możesz np przekazywać jako parametr szerokość (zamiast 640 na stałe) i użyć nazw typu 'isMatched', 'setIsMatched'. I w momencie kiedy wykorzystujesz hooka nadać zwracanej wartości odpowiednią nazwę:
`
const isMobile = useScreenWidth(640);
`

Reply:
chociaż to takie nice to have, tutaj i tak prawdopodobnie nie użyjesz wiele razy tego hooka, ale warto mieć z tyłu głowy, że w przypadku dużego projektu takie bardziej generyczne rozwiązanie byłoby preferowane :D